R/PeCorA_preprocessing.R

```diff
@@ -0,0 +1,65 @@
+#' @title PeCorA_preprocessing
+#' @description steps to pre-process data for PeCorA
+#' @param t data in PeCorA format
+#' @param area_column_name column number with peak areas
+#' @param threshold_to_filter threshould value of peak areas to filter
+#' @param control_name control reference
+#' @author Maria Dermit <maria.dermit@qmul.ac.uk>
+#' @return dataframe output of pre-process data ready for PeCorA analalysis
+#' @details Peak areas are log transformed and scaled to center.After global scaling, each peptide scaling was performed to center each peptide relative to the mean of the control group’s peak area.
+#' @examples
+#' \dontrun{
+#' if(interactive()){
+#'  scaled_peptides <- PeCorA_preprocessing(t)
+#'  }
+#' }
+#' @rdname PeCorA_preprocessing
+#' @importFrom standardize scale_by
+#' @importFrom utils txtProgressBar setTxtProgressBar
+#' @importFrom stats p.adjust
+#' @export
+PeCorA_preprocessing <- function (t,area_column_name,threshold_to_filter,control_name){
+  # make new column that combines petide and charge
+  t["modpep_z"] = paste(t$Peptide.Modified.Sequence, "all", sep = "_")
+  # make column that combines condition and replicate
+  t["condition_rep"] = paste(t$Condition, t$BioReplicate, sep = "_")
+  n_reps_total <- length(unique(t$condition_rep))
+  ## filter the data to remove NA, <100 area, and then reps without all measures
+  if(length(which(is.na(as.numeric(t[,area_column_name]))==TRUE))>0){
+    t <- t[-suppressWarnings(which(is.na(as.numeric(t[,area_column_name]))==TRUE)),]
+  }
+  t <- t[-which(as.numeric(t[,area_column_name])<=threshold_to_filter),]
+  t <- t[t$modpep_z %in% names(which(table(t$modpep_z)==n_reps_total)),]
+
+
+  ## scale the data and then plot
+  t$ms1log2<-log2(as.numeric(t[,area_column_name]))
+  t$ms1scaled <- scale_by(ms1log2 ~ Condition*BioReplicate, t)
```

Comment:
does this need to be changed to?
t$ms1scaled <- standardize::scale_by(ms1log2 ~ Condition*BioReplicate, t)

Reply:
Changed